Server/BackEndMonitoringServer/src/CProcessInfo.cpp

```diff
@@ -0,0 +1,186 @@
+        m_cpu_usage(other.m_cpu_usage),
+        m_ram_usage(other.m_ram_usage),
+        m_pagefile_usage(other.m_pagefile_usage),
+        m_count_type(other.m_count_type),
+        m_last_kernel_time(other.m_last_kernel_time),
+        m_last_sys_time(other.m_last_sys_time),
+        m_last_user_time(other.m_last_user_time),
+        m_is_initialized(other.m_is_initialized)
+{ }
+
+CProcessInfo& CProcessInfo::operator=(const CProcessInfo& other)
+{
+    m_PID = other.m_PID;
+    m_count_of_processors = other.m_count_of_processors;
+    m_cpu_usage = other.m_cpu_usage;
+    m_ram_usage = other.m_ram_usage;
+    m_pagefile_usage = other.m_pagefile_usage;
+    m_count_type = other.m_count_type;
+    m_last_kernel_time = other.m_last_kernel_time;
+    m_last_sys_time = other.m_last_sys_time;
+    m_last_user_time = other.m_last_user_time;
+    m_is_initialized = other.m_is_initialized;
+    return *this;
+}
+
+bool CProcessInfo::Initialize()
+{
+    if(m_is_initialized)
+    {
+        return false;
+    }
+    bool success;
+    
+    success = PlatformUtils::GetProcessTimes(m_PID, m_last_sys_time, 
+                                             m_last_kernel_time,
+                                             m_last_user_time);
+    if (!success)
+    { return success;}
+
+    success = PlatformUtils::GetProcessMemoryUsage(m_PID, m_ram_usage, 
+                                                   m_pagefile_usage);
+    m_is_initialized = success;
+    return success;
+}
+
+bool CProcessInfo::TryToUpdateCurrentStatus()
+{
+    if(!m_is_initialized)
+    {
+        return false;
+    }
+    bool success;
+
+    success = CountCpuUsage();
+    if (!success)
+    { return success;}
+
+    success = PlatformUtils::GetProcessMemoryUsage(m_PID, m_ram_usage,
+                                                   m_pagefile_usage);
+    return success;
+}
+
+bool CProcessInfo::IsActive() const
+{
+    if (!m_is_initialized)
+    {
+        return false;
+    }
+
+    return PlatformUtils::CheckIsProcessActive(m_PID);
+}
+
+bool CProcessInfo::GetPID(unsigned& value) const
+{   
+    if (m_is_initialized)
+    {
+        value = m_PID;
+        return true;
+    }
+    return false;
+}
+
+bool CProcessInfo::GetCpuUsage(double& value) const
+{
+    if (m_is_initialized)
+    {
+        value = m_cpu_usage;
+        return true;
+    }
+    return false;
+}
+
+bool CProcessInfo::GetRamUsage(long double& value) const
+{
+    if (m_is_initialized)
+    {
+        value = Utils::ConvertToCountType(m_ram_usage, m_count_type);
+        return true;
+    }
+    return false;
+}
+
+bool CProcessInfo::GetPagefileUsage(long double& value) const
+{
+    if (m_is_initialized)
+    {
+        value = Utils::ConvertToCountType(m_pagefile_usage, m_count_type);
+        return true;
+
+    }
+    return false;
+}
+
+EMemoryConvertType CProcessInfo::GetMemoryCountType() const
+{ return m_count_type;}
+
+
+bool CProcessInfo::CountCpuUsage()
+{
+    bool success;
```

Comment:
By default, bool should be initialized when declared

Reply:
Fixed